tests/test_e2e.py

```diff
@@ -0,0 +1,419 @@
+"""Tests for the Elasticsearch integration initialization."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from unittest import mock
+from unittest.mock import MagicMock
+
+import pytest
+from custom_components.elasticsearch.config_flow import ElasticFlowHandler
+from freezegun.api import FrozenDateTimeFactory
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt as dt_util
+from pytest_homeassistant_custom_component.common import (
+    MockConfigEntry,  # noqa: F401
+)
+from syrupy.assertion import SnapshotAssertion
+
+from tests import const
+
+if TYPE_CHECKING:
+    from homeassistant.core import HomeAssistant
+
+MODULE = "custom_components.elasticsearch"
+
+
+@pytest.fixture(autouse=True)
+def _fix_system_info():
+    """Return a mock system info."""
+    with mock.patch("custom_components.elasticsearch.es_publish_pipeline.SystemInfo") as system_info:
+        system_info_instance = system_info.return_value
+        system_info_instance.async_get_system_info = mock.AsyncMock(
+            return_value=mock.Mock(
+                version="1.0.0",
+                arch="x86",
+                os_name="Linux",
+                hostname="my_es_host",
+            ),
+        )
+
+        yield
+
+
+def strip_headers_from_mock_calls(mock_calls):
```

Comment:
Consider adding this as a part of a get_calls(include_headers=False) as a function to es_mocker